upgrade-guides/v4.0-v5.0.md

```diff
@@ -0,0 +1,42 @@
+# How to Upgrade Hazelcast-wm Version 4.0 to Version 5.0
+
+This guides summarizes what you need to know if you are upgrading from Hazelcast-wm 4.0 version to 5.0 version.
+
+
+## Hazelcast-wm supports Hazelcast 5.x
+
+Hazelcast-wm supported Hazelcast 4.x versions, whereas Hazelcast-wm supports Hazelcast 5.x. One notable change in Hazelcast 5.x 
+that can affect Hazelcast-wm usage is that Hazelcast clients have indefinite cluster connect timeout by default,
+which was previously 20 seconds. For more info on the topic, see
+[this page](https://docs.hazelcast.com/hazelcast/5.4/clients/java#configuring-client-connection-retry)
+
+To preserve the 20 seconds cluster timeout behaviour in Hazelcast 5.x clients, you need to add this configuration to your 
+hazelcast-client.xml config:
+
+```xml
+<connection-strategy>
+        <connection-retry>
+            <cluster-connect-timeout-millis>20000</cluster-connect-timeout-millis>
+        </connection-retry>
+    </connection-strategy>
+```
+
+## Dependencies are upgraded
+
+Hazelcast-wm is tested with:
+
+- JSP API 2.3+
+- Servlet API 4.0+ 
+- Tomcat 9.0.z
+- Eclipse Jetty 10.0.z
+- Spring 5.3
+- Spring security 5.8
+
+## hazelcast:hazelcast-all does not exist anymore in 5.x
+
+Use artifact hazelcast instead. 
+
+## Upgrade your Hazelcast-wm if using clients
+
+If they're using `use-client=true` and any currently released HZ 5.x then you should remove hazelcast-wm-4.0.jar from the
```

Comment:
@olukas this is from https://hazelcast.atlassian.net/browse/HZG-17?focusedCommentId=98670 but I didn't quite understand what you meant, so can you explain what you mean?

Reply:
IIUC (from discussion with @frant-hartm ) `hazelcast-wm-4.0.jar` should be updated to `hazelcast-wm-5.0.jar` in all HZ distribution which don't contain it (i.e. all which were released before hazelcast-wm-5.0). 

If `use-client=true` is used then client can connect to remote HZ (independent on Spring Boot or so) and for that server `hazelcast-wm` should be updated.

Reply:
> If use-client=true is used then client can connect to remote HZ (independent on Spring Boot or so) and for that server hazelcast-wm should be updated.

When I think about it, I feel like what we call "server" is just a hazelcast cluster, which don't need hazelcast-wm, just a regular cluster. Am I wrong in this? So there should be no need to upgrade remote

Reply:
ah, yes, you're right. I totally got lost there :) feel free to remove this part